File: Difficulty: Easy/Segregate 0s and 1s/segregate-0s-and-1s.py
class Solution:
    def segregate0and1(self, arr):
        # code here
        n = len(arr)
        l, r = 0, n-1
        while(l < r):
            while(l < r and arr[l] == 0):
                l += 1
            while(r > l and arr[r] == 1):
                r -= 1
            
            arr[l], arr[r] = arr[r], arr[l]
            l += 1
            r -= 1
        return arr

File: Difficulty: Easy/Segregate 0s and 1s/segregate-0s-and-1s.java
class Solution {
    void segregate0and1(int[] arr) {
        // code here
        int n = arr.length;
        int count = 0;
        for(int i : arr){
            if(i == 0){
                count++;
            }
        }
        for(int i = 0; i < n; i++){
            if(i < count){
                arr[i] = 0;
            }else{
                arr[i] = 1;
            }
        }
        
    }
}
